Default value for scriptTelemetry

pooneh-nb opened the pull request:
#### 5fac53f4db17e5501dd5c47646f7f85ebf7a9d5f
<pre>
Default value for scriptTelemetry
<a href="https://bugs.webkit.org/show_bug.cgi?id=284778">https://bugs.webkit.org/show_bug.cgi?id=284778</a>
<a href="https://rdar.apple.com/141566635">rdar://141566635</a>

Reviewed by NOBODY (OOPS!).

Return default values for scriptTelemetry for specific elements.

* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::value const):
* Source/WebCore/page/ScriptTelemetryCategory.cpp:
(WebCore::makeLogMessage):
(WebCore::telemetryString):
* Source/WebCore/page/ScriptTelemetryCategory.h:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5fac53f4db17e5501dd5c47646f7f85ebf7a9d5f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/81595 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/1121 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/35543 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/86130 "Hash 5fac53f4 for PR 38034 does not build (failure)") | [❌ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/32589 "Hash 5fac53f4 for PR 38034 does not build (failure)") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/83705 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1140 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/8939 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/86130 "Hash 5fac53f4 for PR 38034 does not build (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/59/builds/32589 "Hash 5fac53f4 for PR 38034 does not build (failure)") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/84664 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/807 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74254 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/86130 "Hash 5fac53f4 for PR 38034 does not build (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/704 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/28433 "Passed tests") | [❌ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/31044 "Hash 5fac53f4 for PR 38034 does not build (failure)") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/72140 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29027 "Passed tests") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/87570 "Hash 5fac53f4 for PR 38034 does not build (failure)") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/8833 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/6262 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/87570 "Hash 5fac53f4 for PR 38034 does not build (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9012 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70081 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/87570 "Hash 5fac53f4 for PR 38034 does not build (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15300 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/14217 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/8788 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/14316 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/8628 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12149 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10435 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

sysrqb commented on Source/WebCore/page/ScriptTelemetryCategory.cpp:
```diff
@@ -83,4 +83,11 @@ String makeLogMessage(const URL& url, ScriptTelemetryCategory category)
 
+#if USE(APPLE_INTERNAL_SDK) && __has_include(<WebKitAdditions/ScriptTelemetryCategoryAdditions.cpp>)
+#else
+String telemetryString(const HTMLInputElement&)
```
Can you put this into the `#else` above?

pooneh-nb committed:
Default value for scriptTelemetry
https://bugs.webkit.org/show_bug.cgi?id=284778
rdar://141566635

Reviewed by NOBODY (OOPS!).

Return default values for scriptTelemetry for specific elements.

* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::value const):
* Source/WebCore/page/ScriptTelemetryCategory.cpp:
(WebCore::makeLogMessage):
(WebCore::telemetryString):
* Source/WebCore/page/ScriptTelemetryCategory.h:

rniwa commented on Source/WebCore/page/ScriptTelemetryCategory.h:
```diff
@@ -24,6 +24,7 @@
  */
 
 #pragma once
+#include <WebCore/HTMLInputElement.h>
```
Why do we need to include HTMLInputElement.h here.
A forward declaration of `HTMLInputElement` should be sufficient.

pooneh-nb replied:
We need it because we call its functions internally:
[requests](rdar://141566635)

whsieh replied:
I believe Ryosuke is suggesting that we can just add a forward declaration to the header:

```
class HTMLInputElement;
```

...and then `#include "HTMLInputElement.h"` in the implementation file. (There also shouldn't be angle brackets with `<WebCore/` since we're already in the WebCore subproject).
